src/Corrade/Utility/Sha1.cpp

```diff
@@ -92,28 +99,34 @@ Sha1& Sha1::operator<<(const std::string& data) {
 #pragma GCC optimize ("O2")
 #endif
 Sha1::Digest Sha1::digest() {
-    /* Add '1' bit to the leftovers, pad to (n*64)+56 bytes */
-    _buffer.append(1, '\x80');
-    _buffer.append((_buffer.size() > 56 ? 120 : 56) - _buffer.size(), 0);
+    /* Add '1' bit to the leftovers */
+    _buffer[_bufferSize++] = '\x80';
+
+    /* Pad to (n*64)+56 bytes */
+    const size_t padding = (_bufferSize > 56 ? 120 : 56) - _bufferSize;
```

Comment:
`CORRADE_INTERNAL_ASSERT(_bufferSize + padding + 8 <= sizeof(_buffer))`? Since we don't have a string growing indefinitely anymore, it's important we don't overflow :)